aspnet/security/cross-site-scripting.rst

```diff
@@ -1,8 +1,256 @@
+of the web page through DOM manipulation or redirect the browser to another page. XSS vulnerabilities generally occur when an application takes user 
+input and outputs it in a page without validating, encoding or escaping it.
 
-.. _issue: https://github.com/aspnet/Docs/issues/90
\ No newline at end of file
+.. contents:: Sections:
+  :local:
+  :depth: 1
+
+Protecting your application against XSS
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+At a basic level XSS works by tricking your application into inserting a ``<script>`` tag into your rendered page, or by inserting an ``On*`` event 
+into an element. Developers should use the following prevention steps to avoid introducing XSS into their application.
+
+1. Never put untrusted data into your HTML input, unless you follow the rest of the steps below. 
+   Untrusted data is any data that may be controlled by an attacker, HTML form inputs, query strings,
+   HTTP headers, even data sourced from a database as an attacker may be able to breach your database even if they cannot breach your application.
+#. Before putting untrusted data inside an HTML element ensure it is HTML encoded. HTML encoding takes characters such as < and 
+   changes them into a safe form like &lt;
+#. Before putting untrusted data into an HTML attribute ensure it is HTML attribute encoded. HTML attribute encoding is a superset of
+   HTML encoding and encodes additional characters such as " and \'.
+#. Before putting untrusted data into JavaScript place the data in an HTML element whose contents you retrieve at runtime.
+   If this is not possible then ensure the data is JavaScript encoded. JavaScript encoding takes dangerous characters for JavaScript
+   and replaces them with their hex, for example < would be encoded as ``\u003C``.
+#. Before putting untrusted data into a URL query string ensure it is URL encoded.
+
+
+
+HTML Encoding using Razor
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Razor engine used in MVC automatically encodes all output sourced from variables, unless you work really hard to prevent it doing so.
+It uses HTML Attribute encoding rules whenever you use the `@` directive. As HTML attribute encoding is a superset of HTML encoding this
+means you don't have to concern yourself with whether you should use HTML or HTML attribute encoding, you just need to ensure that you
+only use @ in an HTML context, not when attempting to insert untrusted input directly into JavaScript. Tag helpers will also encode 
+input you use in tag parameters.
+
+Take the following Razor view;
+
+:: 
+
+  @{
+      var untrustedInput = "<\"123\">";
+  }
+
+  @untrustedInput
+
+This view outputs the contents of the `untrustedInput` variable. This variable includes some characters which are using in XSS attacks, namely
+<, " and >. If you look at the source for the rendered page you will see the characters have been encoded. The source for the page will be
```

Comment:
~~If you look at the source for the rendered page you will see the characters have been encoded. The source for the page will be~~

The rendered output will be encoded  as:

or

Examining the source shows the rendered output encoded as: